src/main/java/net/openhft/compiler/CachedCompiler.java

```diff
@@ -77,67 +90,58 @@ public void close() {
         }
     }
 
+    /**
+     * Delegates to {@link #loadFromJava(ClassLoader, String, String, PrintWriter, DiagnosticListener)}.
+     * <ul>
+     *   <li>The class loader of {@link CachedCompiler} is used for defining and loading the class
+     *   <li>Only error diagnostics are collected, and are written to {@link System#err}
+     * </ul>
+     */
     public Class<?> loadFromJava(@NotNull String className, @NotNull String javaCode) throws ClassNotFoundException {
         return loadFromJava(getClass().getClassLoader(), className, javaCode, DEFAULT_WRITER);
     }
 
+    /**
+     * Delegates to {@link #loadFromJava(ClassLoader, String, String, PrintWriter, DiagnosticListener)}.
+     * Only error diagnostics are collected, and are written to {@link System#err}.
+     */
     public Class<?> loadFromJava(@NotNull ClassLoader classLoader,
                               @NotNull String className,
                               @NotNull String javaCode) throws ClassNotFoundException {
         return loadFromJava(classLoader, className, javaCode, DEFAULT_WRITER);
     }
 
-    @NotNull
-    Map<String, byte[]> compileFromJava(@NotNull String className, @NotNull String javaCode, MyJavaFileManager fileManager) {
-        return compileFromJava(className, javaCode, DEFAULT_WRITER, fileManager);
-    }
-
-    @NotNull
-    Map<String, byte[]> compileFromJava(@NotNull String className,
```

Comment:
I have moved this method down a few lines so that all `loadFromJava` methods are grouped together.